### DISCUSSION TITLE
Extending task_name_prefix with support for role names

### DISCUSSION THREAD

**Discussion by CarstenGrohmann** [ansible/ansible-lint#3197] [Ideas]
Currently Ansible-lint is able to check if task names start with the base name of the current task file by using the `stem` pattern in `task_name_prefix`.

What do you think about adding the role name with the pattern `rolename`?

Also, I think that `stem` is the right technical term, but with us humans in mind, another term like `taskfile` would make more sense. What do you think about renaming `stem` to `taskfile` or adding `taskfile` and marking `stem` as deprecated?

Regards,
Carsten

**Comment by aehliglucas** [comment]
Created https://github.com/ansible/ansible-lint/pull/3221 as a proposal on how to solve this.
